hilti/toolchain/src/compiler/visitors/printer.cc

```diff
@@ -984,6 +984,15 @@ void hilti::detail::printAST(const Node& root, std::ostream& out, bool compact)
 void hilti::detail::printAST(const Node& root, printer::Stream& stream) {
     util::timing::Collector _("hilti/printer");
 
+    if ( auto t = root.tryAs<Type>() ) {
+        if ( ! stream.isExpandSubsequentType() ) {
+            if ( auto id = t->typeID() ) {
+                stream << *id;
+                return;
```

Comment:
Did you perhaps here mean:
```
        if ( ! stream.isExpandSubsequentType() ) {
            if ( auto id = t->typeID() ) {
                stream << *id;
            }
            return;
        }
```
in case the ->typeID() call returns something that coerces to false?

Reply:
No, it's correct as written: a type may or may not have a type ID. If it has one, we want to print it (unless explicitly asked not to). If it doesn't, we fall back to the remainder of the method, which will print the full type. With your suggested change, we wouldn't print anything at all in that case, which would leave a gap in the output.

But valid question, thanks for watching!